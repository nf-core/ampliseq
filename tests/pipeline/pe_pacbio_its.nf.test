nextflow_pipeline {

    name "Test Workflow main.nf"
    script "main.nf"
    options "-profile test_pacbio_its,docker"
    tag "pipeline"

    test("Pacbio ITS") {

        when {
            params {
                outdir = "$outputDir"
            }
        }

        then {
            assertAll(
                { assert workflow.success },
                { assert snapshot(path("$outputDir/pipeline_info/software_versions.yml")).match("software_versions") },
                { assert snapshot(path("$outputDir/overall_summary.tsv")).match("overall_summary_tsv") },
                { assert snapshot(path("$outputDir/assignsh/ASV_tax_species_SH.tsv")).match("assignsh") },
                { assert new File("$outputDir/assignsh/ASV_cut.vsearch.txt").exists() },
                { assert snapshot(path("$outputDir/barrnap/rrna.arc.gff"),
                                  path("$outputDir/barrnap/rrna.bac.gff"),
                                  path("$outputDir/barrnap/rrna.euk.gff"),
                                  path("$outputDir/barrnap/rrna.mito.gff")).match("barrnap") },
                { assert new File("$outputDir/barrnap/summary.tsv").exists() },
                { assert snapshot(path("$outputDir/cutadapt/cutadapt_summary.tsv")).match("cutadapt") },
                { assert new File("$outputDir/cutadapt/pb1.trimmed.cutadapt.log").exists() },
                { assert new File("$outputDir/cutadapt/pb2.trimmed.cutadapt.log").exists() },
                { assert new File("$outputDir/cutadapt/pb3.trimmed.cutadapt.log").exists() },
                { assert snapshot(path("$outputDir/dada2/ASV_seqs.fasta"),
                                  path("$outputDir/dada2/ASV_table.tsv"),
                                  path("$outputDir/dada2/ref_taxonomy.txt"),
                                  path("$outputDir/dada2/DADA2_stats.tsv"),
                                  path("$outputDir/dada2/DADA2_table.rds"),
                                  path("$outputDir/dada2/DADA2_table.tsv")).match("dada2") },
                { assert new File("$outputDir/dada2/ASV_tax.tsv").exists() },  
                { assert new File("$outputDir/dada2/ASV_tax_species.tsv").exists() },
                { assert new File("$outputDir/dada2/ASV_ITS_tax.tsv").exists() },  
                { assert new File("$outputDir/dada2/ASV_ITS_tax_species.tsv").exists() },
                { assert new File("$outputDir/fastqc/pb1_fastqc.html").exists() },
                { assert new File("$outputDir/fastqc/pb2_fastqc.html").exists() },
                { assert new File("$outputDir/fastqc/pb3_fastqc.html").exists() },
                { assert snapshot(path("$outputDir/input/Samplesheet_pacbio_ITS.tsv"),
                                  path("$outputDir/input/Metadata_pacbio_ITS.tsv")).match("input") },
                { assert snapshot(path("$outputDir/itsx/ASV_ITS_seqs.full.fasta"),
                                  path("$outputDir/itsx/ITSx.args.txt")).match("itsx") },
                { assert new File("$outputDir/itsx/ASV_ITS_seqs.summary.txt").exists() },                  
                { assert snapshot(path("$outputDir/multiqc/multiqc_data/multiqc_fastqc.txt"),
                                  path("$outputDir/multiqc/multiqc_data/multiqc_general_stats.txt"),
                                  path("$outputDir/multiqc/multiqc_data/multiqc_cutadapt.txt")).match("multiqc") },
                { assert snapshot(path("$outputDir/SBDI/dna.tsv"),
                                  path("$outputDir/SBDI/emof.tsv"),
                                  path("$outputDir/SBDI/event.tsv")).match("SBDI") },
                { assert new File("$outputDir/SBDI/annotation.tsv").exists() },
                { assert new File("$outputDir/SBDI/asv-table.tsv").exists() }                                   
            )
        }
    }
}
