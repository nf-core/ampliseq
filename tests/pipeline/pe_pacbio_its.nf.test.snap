{
    "assignsh": {
        "content": [
            "ASV_tax_species_SH.tsv:md5,91211af4f17cf4537fe715b50a0b2b64"
        ],
        "timestamp": "2023-04-27T23:11:54+0000"
    },
    "input": {
        "content": [
            "Samplesheet_pacbio_ITS.tsv:md5,905b0fc4850149d6457c9da7076c2c2d",
            "Metadata_pacbio_ITS.tsv:md5,51d81b0fe29f6377fa21ce8399fc5ae0"
        ],
        "timestamp": "2023-04-27T23:11:54+0000"
    },
    "itsx": {
        "content": [
            "ASV_ITS_seqs.full.fasta:md5,dab051f759c877a79adf47f5729f5665",
            "ITSx.args.txt:md5,32430f2c36d83c08273352bad45857e8"
        ],
        "timestamp": "2023-04-27T23:11:54+0000"
    },
    "SBDI": {
        "content": [
            "dna.tsv:md5,061ca46725294da75acd8237390f4291",
            "emof.tsv:md5,c6599c76a70fe5458db56ff8c2e89a37",
            "event.tsv:md5,bfcf7d8ac2a3c6b48d075d6dc235f0e4"
        ],
        "timestamp": "2023-04-27T23:11:54+0000"
    },
    "software_versions": {
        "content": [
            "software_versions.yml:md5,9327d405e8183876caa2220e1b320a57"
        ],
        "timestamp": "2023-04-27T23:11:54+0000"
    },
    "cutadapt": {
        "content": [
            "cutadapt_summary.tsv:md5,d185ccd02c22b859fcaf2e5158ce70c6"
        ],
        "timestamp": "2023-04-27T23:11:54+0000"
    },
    "overall_summary_tsv": {
        "content": [
            "overall_summary.tsv:md5,bdeeffdffcbd7c040a8f94ec254133fe"
        ],
        "timestamp": "2023-04-27T23:11:54+0000"
    },
    "dada2": {
        "content": [
            "ASV_seqs.fasta:md5,a2632b1cc533d290db19ceedaf350dcd",
            "ASV_table.tsv:md5,6de5507a0fcb0c9b03f8fec0270440be",
            "ref_taxonomy.txt:md5,7151eab71d460eeb86061d2e4a16bafc",
            "DADA2_stats.tsv:md5,fd9a2c76b56cf5094fd93e7717441165",
            "DADA2_table.rds:md5,a36558c04a7b21050bb6e479b09bd483",
            "DADA2_table.tsv:md5,7e7aab7ae9a3b20fb08f3fe7f8b0430c"
        ],
        "timestamp": "2023-04-27T23:11:54+0000"
    },
    "barrnap": {
        "content": [
            "rrna.arc.gff:md5,6bb50a4c244d80d56e759ecfbb045381",
            "rrna.bac.gff:md5,7c38dd6692b792ace17142158afeb33e",
            "rrna.euk.gff:md5,8ae155cbe43335084a0c9a2ccd655cb3",
            "rrna.mito.gff:md5,df19e1b84ba6f691d20c72b397c88abf"
        ],
        "timestamp": "2023-04-27T23:11:54+0000"
    },
    "multiqc": {
        "content": [
            "multiqc_fastqc.txt:md5,d289cd1a2523f9ad66e93a895dcbe52c",
            "multiqc_general_stats.txt:md5,05682be32bc30bc4610f0ca608cafe67",
            "multiqc_cutadapt.txt:md5,1b3b6833e78db31ab12e5c16b7fa1d73"
        ],
        "timestamp": "2023-04-27T23:11:54+0000"
    }
}